src/pages/UserPage.tsx

```diff
@@ -0,0 +1,120 @@
+import { useSelector } from "react-redux";
+import { RootState } from "../redux/store";
+import {
+  Card,
+  CardContent,
+  Grid,
+  Typography,
+  CircularProgress,
+  Container,
+  Box,
+} from "@mui/material";
+import { Link } from "react-router-dom";
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import { fetchUser } from "../redux/user/user.actions";
+import { useEffect } from "react";
+import { useAppDispatch } from "../redux/hooks";
+import { Linkify } from "../utils/utilities";
+import { theme } from "../styles/theme";
+
+const defaultTheme = createTheme(theme);
+
+export const UserPage = (): JSX.Element => {
+  const dispatch = useAppDispatch();
+  const auth = useSelector((state: RootState) => state.auth);
+  const user = useSelector((state: RootState) => state.users);
```

Comment:
Since you're using user.user a lot, why don't you deconstruct this to make the syntax shorter and easier to read.

`const {user} = useSelector...` so you can simply call user.posts, etc.